rim/tests.py

```diff
@@ -1,3 +1,39 @@
-from django.test import TestCase
+from django.test import TestCase, Client
 
 # Create your tests here.
+class TestHttpResponse(TestCase):
+    # Url dictionary to hold names for existing rim web pages, add as neccessary 
+    urlDictionary = {
+        "Admin Panel": '/admin/',
+        "Home": '/',
+        "Add": '/add/',
+        "Edit": '/edit/1',
+        "Client": '/client/1'
+    }
+
+    # get_key looks up they key in urlDictionary based on the provided value
+    def get_key(self, val):
+        for key, value in self.urlDictionary.items():
+            if val == value:
+                return key
+
+    # test_http_response iterates through the pages and checks if all pages all returning HTTP 200(OK) 
+    # response, otherwise the test will be failed and the name/returned code of the faulty page will be printed to the console
+    # and a ConnectionError will be raised
+    def test_http_response(self):
+        client = Client()
+        passedFlag = True
+        for url in self.urlDictionary.values():
+            try:
+                response = client.post(url, follow=True)
```

Comment:
Also, since the client isn't logged in, every response is just a redirect to the login page. That's still a worthwhile test to make sure all of our pages require the user to be logged in, but we should probably also test things with a valid, logged in user. I recommend creating a user and [logging in](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.Client.login) in the `setUp()` method.